### DISCUSSION TITLE
File upload from local computer (for notebooks on WASM or community cloud)

### DISCUSSION THREAD

**Discussion by pepa8004** [marimo-team/marimo#3064] [Q&A]
Hi,

Is it possible for the app in WASM or the community cloud to upload a file from a disk on the local computer?

I have a CSV file containing some data that the app should then process (performing some statistical calculations and visualizations).

I tried using mo.ui.file_browser. This works fine when running locally, but when I put the notebook on WASM, I see a different filesystem, not my local hard drive.

Is there a solution so that the app/notebook on WASM or the community cloud can browse the local disk (e.g., using mo.ui.file_browser or some other method) and select data to process?

Thanks in advance.

**Comment by mscolnick** [comment]
You can use `mo.ui.file()` and let the user upload the file themselves. WASM notebooks don't have permissions to look at the user's files natively (which is intentional for security)
